src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -1641,33 +1641,35 @@ class spell_sha_nature_guardian : public SpellScriptLoader
             return ValidateSpellInfo({ SPELL_SHAMAN_NATURE_GUARDIAN });
         }
 
-        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        bool CheckProc(ProcEventInfo& eventInfo)
         {
-            PreventDefaultAction();
             DamageInfo* damageInfo = eventInfo.GetDamageInfo();
             if (!damageInfo || !damageInfo->GetDamage())
-                return;
-
-            int32 healthpct = aurEff->GetSpellInfo()->Effects[EFFECT_1].CalcValue(); // %s2 - the 30% threshold for health
+                return false;
 
+            int32 healthpct = GetSpellInfo()->Effects[EFFECT_1].CalcValue();
             if (Unit* target = eventInfo.GetActionTarget())
-            {
                 if (target->HealthBelowPctDamaged(healthpct, damageInfo->GetDamage()))
-                {
-                    CastSpellExtraArgs args(aurEff);
-                    args.AddSpellBP0(CalculatePct(target->GetMaxHealth(), aurEff->GetAmount()));
-                    target->CastSpell(target, SPELL_SHAMAN_NATURE_GUARDIAN, args);
-
-                    // Threat reduction is around 10% confirmed in retail and from wiki
-                    Unit* attacker = eventInfo.GetActor();
-                    if (attacker->IsAlive())
-                        attacker->GetThreatManager().ModifyThreatByPercent(target, -10);
-                }
-            }
+                    return true;
+
+            return false;
+        }
+
+        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        {
+            PreventDefaultAction();
+
+            Unit* target = eventInfo.GetActionTarget();
+            CastSpellExtraArgs args(aurEff);
+            args.AddSpellBP0(CalculatePct(target->GetMaxHealth(), aurEff->GetAmount()));
+            target->CastSpell(target, SPELL_SHAMAN_NATURE_GUARDIAN, args);
+            if (Unit* attacker = eventInfo.GetActor())
```

Comment:
@ariel- here can be null right?
After recent changes, proc actor can be offline player DOT?

Reply:
Yup it can